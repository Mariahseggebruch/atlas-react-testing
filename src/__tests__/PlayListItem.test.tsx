// PlayListItem.test.tsx
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import PlayListItem from '../components/PlayListItem';

it('renders PlayListItem with all details', () => {
  const { container } = render(
    <PlayListItem title="Song Title" artist="Artist Name" length="3:45" />
  );
  expect(container).toMatchSnapshot();
});

it('renders PlayListItem with missing details', () => {
  const { container } = render(
    <PlayListItem title="Song Title" artist="" length="" />
  );
  expect(container).toMatchSnapshot();
});