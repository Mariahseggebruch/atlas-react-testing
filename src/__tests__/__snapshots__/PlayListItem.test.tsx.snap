// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders PlayListItem with all details 1`] = `
<div>
  <div
    class="w-full mb-2 pr-3 flex justify-between font-medium"
  >
    <div
      class="wrapper text-left text-sm"
    >
      <div
        class="text-base"
      >
        Song Title
      </div>
      <div
        class="text-butterfly-bush-950"
      >
        Artist Name
      </div>
    </div>
    <div
      class="length flex items-center"
    >
      <div
        class="text-sm text-butterfly-bush-950"
      >
        3:45
      </div>
    </div>
  </div>
</div>
`;

exports[`renders PlayListItem with missing details 1`] = `
<div>
  <div
    class="w-full mb-2 pr-3 flex justify-between font-medium"
  >
    <div
      class="wrapper text-left text-sm"
    >
      <div
        class="text-base"
      >
        Song Title
      </div>
      <div
        class="text-butterfly-bush-950"
      />
    </div>
    <div
      class="length flex items-center"
    >
      <div
        class="text-sm text-butterfly-bush-950"
      />
    </div>
  </div>
</div>
`;
